### ISSUE TITLE
Bazel swallows credential helper standard error output instead to echoing it to the user

### ISSUE THREAD

**Issue by hochhaus** [bazelbuild/bazel#20968]
### Description of the bug:

The credential helper [proposal](https://github.com/bazelbuild/proposals/blob/main/designs/2022-06-07-bazel-credential-helpers.md) states:

> A credential helper may provide a human-readable error message in its standard error that Bazel will print.

Instead bazel swallows stderr output from the credential helper output. This combined with ignored credential helper return codes (#20146) results in a relatively poor end user experience in credential helper failure cases.

Ideally bazel would echo stderr output to the user immediately (even in non-error cases) such that the credential helper could provide instructions to the user (eg: insert or touch your yubikey) even before the credential helper exits.

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Add a credential helper which sends a user message to stderr and returns a failure code. For example:

```
#!/usr/bin/env python3

import sys

print('output for the end user', file=sys.stderr)
sys.exit(1)
```

### Which operating system are you running Bazel on?

Debian Linux 12 Bookworm

### What is the output of `bazel info release`?

release 6.4.0

**Comment by Wyverald** [comment]
This is a dupe of #20146, which is fixed in https://github.com/bazelbuild/bazel/commit/f052e79453afeefc46474641b3010226a3778700. It's not backported to 6.4.0 (or 7.0.0, for that matter...). I've opened a backport request for 7.1.0 (https://github.com/bazelbuild/bazel/issues/20983).

**Comment by hochhaus** [comment]
Thanks @Wyverald. Unfortunately, #20146 is talking about the error code returned from the credential helper being ignore. This issue is instead talking about std err not being echoed to the user.

Is it possible for this issue to be reopened?

**Comment by Wyverald** [comment]
> It is desirable for stderr to be echoed to the user even in the case that the credential helper is still running (eg: "please insert your yubikey").

That sounds like a feature request all by itself.

@tjgq was interactive usage of the credentials helpers ever planned?

**Comment by hochhaus** [comment]
> That sounds like a feature request all by itself.

Fair enough. That said all that is required for this use case is echoing stderr and that is explicitly mentioned in the [proposal](https://github.com/bazelbuild/proposals/blob/main/designs/2022-06-07-bazel-credential-helpers.md):

> A credential helper may provide a human-readable error message in its standard error that Bazel will print.
